src/components/capture/GenerateButton.tsx

```diff
@@ -84,9 +84,18 @@ function CaptureGenerateButton({ disabled, callFailed, subscription }: Props) {
     // Resource Config Store
     const resourceConfig = useResourceConfig_resourceConfig();
 
-    const endpointConfigErrorFlag = editWorkflow
-        ? endpointConfigChanged() && endpointConfigErrorsExist
-        : endpointConfigErrorsExist || isEmpty(endpointConfigData);
+    const endpointConfigErrorFlag = useMemo(() => {
+        return editWorkflow
+            ? (endpointConfigChanged() && endpointConfigErrorsExist) ||
+                  (!draftId && endpointConfigErrorsExist)
+            : endpointConfigErrorsExist || isEmpty(endpointConfigData);
+    }, [
+        draftId,
+        editWorkflow,
+        endpointConfigChanged,
+        endpointConfigData,
+        endpointConfigErrorsExist,
+    ]);
 
```

Comment:
This is mainly for the custom errors. When there is not a draftId and we show the next button if there are errors we need to make sure that the generate button does not submit.